generator/src/templates/partials/apache.hbs

```diff
@@ -0,0 +1,59 @@
+<VirtualHost *:443>
+    ...
+    SSLEngine on
+{{#if (minver "2.4.8" form.serverVersion)}}
+    SSLCertificateFile      /path/to/signed_certificate_followed_by_intermediate_certificates
+{{else}}
+    SSLCertificateFile      /path/to/signed_certificate
+    SSLCertificateChainFile /path/to/intermediate_certificate
+{{/if}}
+    SSLCertificateKeyFile   /path/to/private/key
+
+    # Uncomment the following directive when using client certificate authentication
+    # SSLCACertificateFile    /path/to/ca_certs_for_client_authentication
+    {{#if form.hsts}}
+
+    # HTTP Strict Transport Security (mod_headers is required) ({{output.hstsMaxAge}} seconds)
+    Header always set Strict-Transport-Security "max-age={{output.hstsMaxAge}}"
+    {{/if}}
+    ...
+</VirtualHost>
+
+# {{form.config}} configuration, tweak to your needs
+SSLProtocol             {{#if (minver "2.3.16" form.serverVersion)}}{{replace output.protocols " -SSLv2" ""}}{{else}}{{output.protocols}}{{/if}}
+SSLCipherSuite          {{output.cipherSuites}}
+SSLHonorCipherOrder     on
+{{#if (minver "0.9.8" form.opensslVersion)}}
+  {{#if (matchesminorver "2.2" form.serverVersion)}}
+    {{#if (minver "2.2.24" form.serverVersion)}}
+SSLCompression          off
+    {{/if}}
+  {{/if}}
+  {{#if (matchesminorver "2.4" form.serverVersion)}}
+    {{#if (minver "2.4.3" form.serverVersion)}}
+SSLCompression          off
+    {{/if}}
+  {{/if}}
+{{/if}}
+{{#if (minver "0.9.8f" form.opensslVersion)}}
+  {{#if (matchesminorver "2.2" form.serverVersion)}}
+    {{#if (minver "2.2.30" form.serverVersion)}}
+SSLSessionTickets       off
+    {{/if}}
+  {{/if}}
+  {{#if (matchesminorver "2.4" form.serverVersion)}}
```

Comment:
Or finally you could just create a helper for these two settings to make the template very clear

```
{{#if (apache_ssl_compression_supported form.opensslVersion form.serverVersion)}}
SSLCompression          off
{{/if}}
{{#if (apache_ssl_sessiontickets_supported form.opensslVersion form.serverVersion)}}
SSLSessionTickets       off
{{/if}}
```

Reply:
I really don't want to write helpers specific to specific pieces of software, it makes it very unclear where the logic is actually happening.